### DISCUSSION TITLE
Install problem

### DISCUSSION THREAD

**Discussion by howardt** [elalish/manifold#777] [General]
I tried to do a 'cmake --install .' after a build and it installed things into /usr/local, as I expexted, but it did not install the glm include files in the installation, so that trying to include manifold.h from /usr/loca/include/manifold failed (the crossection include file includes some glm files). Am I doing something wrong?

**Comment by elalish** [comment]
Hmm, @pca006132 do you know what's going on here? Maybe related to #696?

**Comment by pca006132** [comment]
Can you show the complete build log? I suspect that you already have `glm` installed in your system so our script will not install it, see https://github.com/elalish/manifold/blob/master/manifoldDeps.cmake#L49

**Comment by howardt** [comment]
Here's the output. It doesn't seem to find an installed glm. If you need a more verbose log, remind me what flag to give cmake to get it. This is from 'cmake .' using the cached stuff. The options I used in the first build were -DBUILD_SHARED_LIBS=ON and -DMANIFOLD_PAR=TBB . Note that I have separately downloaded and installed oneTBB, as that wasn't installed by your cmake config either (though perhaps that is intentional?).

-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- clipper2 not found, downloading from source
-- GLM: Version 0.9.9.9
-- glm not found, downloading from source
Parallel Backend: TBB
thrust not found, downloading from source
-- Configuring done (0.8s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/howardtrickey/src/manifold/build

When I do a 'sudo cmake --install .' I get this (Note, I've already installed so this mostly just notes that it doesn't have to do it again, but also note that ther is no status noted about glm in any of the include direrectories (though it does note that /usr/local/lib/libglm.dylib is up to date). I believe the glm header tree belongs in /usr/local/include/ . 

-- Install configuration: "Release"
-- Up-to-date: /usr/local/lib/libClipper2.1.3.0.dylib
-- Up-to-date: /usr/local/lib/libClipper2.1.dylib
-- Up-to-date: /usr/local/lib/libClipper2.dylib
-- Up-to-date: /usr/local/include/clipper2/clipper.h
-- Up-to-date: /usr/local/include/clipper2/clipper.version.h
-- Up-to-date: /usr/local/include/clipper2/clipper.core.h
-- Up-to-date: /usr/local/include/clipper2/clipper.engine.h
-- Up-to-date: /usr/local/include/clipper2/clipper.export.h
-- Up-to-date: /usr/local/include/clipper2/clipper.minkowski.h
-- Up-to-date: /usr/local/include/clipper2/clipper.offset.h
-- Up-to-date: /usr/local/include/clipper2/clipper.rectclip.h
-- Up-to-date: /usr/local/lib/pkgconfig/Clipper2.pc
-- Up-to-date: /usr/local/lib/libClipper2.1.3.0.dylib
-- Up-to-date: /usr/local/lib/libClipper2.1.dylib
-- Up-to-date: /usr/local/lib/libClipper2.dylib
-- Up-to-date: /usr/local/include/clipper.h
-- Up-to-date: /usr/local/include/clipper.version.h
-- Up-to-date: /usr/local/include/clipper.core.h
-- Up-to-date: /usr/local/include/clipper.engine.h
-- Up-to-date: /usr/local/include/clipper.export.h
-- Up-to-date: /usr/local/include/clipper.minkowski.h
-- Up-to-date: /usr/local/include/clipper.offset.h
-- Up-to-date: /usr/local/include/clipper.rectclip.h
-- Up-to-date: /usr/local/share/clipper2/clipper2Targets.cmake
-- Up-to-date: /usr/local/share/clipper2/clipper2Targets-release.cmake
-- Up-to-date: /usr/local/lib/libglm.dylib
-- Up-to-date: /usr/local/share/glm/glmTargets.cmake
-- Up-to-date: /usr/local/share/glm/glmTargets-release.cmake
-- Up-to-date: /usr/local/include/manifold/public.h
-- Up-to-date: /usr/local/include/manifold/vec_view.h
-- Up-to-date: /usr/local/include/manifold/collider.h
-- Up-to-date: /usr/local/include/manifold/cross_section.h
-- Up-to-date: /usr/local/include/manifold/polygon.h
-- Installing: /usr/local/lib/libmanifold.2.4.5.dylib
-- Up-to-date: /usr/local/lib/libmanifold.2.dylib
-- Up-to-date: /usr/local/lib/libmanifold.dylib
-- Up-to-date: /usr/local/include/manifold/manifold.h
-- Up-to-date: /usr/local/include/manifold/sdf.h
-- Installing: /usr/local/lib/libmanifoldc.dylib
-- Up-to-date: /usr/local/include/manifoldc.h
-- Up-to-date: /usr/local/include/types.h
-- Up-to-date: /usr/local/share/manifold/manifoldTargets.cmake
-- Up-to-date: /usr/local/share/manifold/manifoldTargets-release.cmake
-- Up-to-date: /usr/local/share/manifold/manifoldDeps.cmake
-- Up-to-date: /usr/local/share/manifold/manifoldConfigVersion.cmake
-- Up-to-date: /usr/local/share/manifold/manifoldConfig.cmake

If I try a simple project that includes "manifold.h" with /usr/local/include/manifold in the include search path, I get:

In file included from /Users/howardtrickey/src/try_manifold/try_manifold.cc:3:
In file included from /usr/local/include/manifold/manifold.h:19:
/usr/local/include/manifold/cross_section.h:21:10: fatal error: 'glm/ext/matrix_float3x2.hpp' file not found
#include "glm/ext/matrix_float3x2.hpp"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

If I download a copy the header files for glm into /usr/local/include/ , then my sample project builds correctly.

**Reply by pca006132** [reply]
I guess we are not installing glm correctly. I will have a look at this at this weekend

**Comment by pca006132** [comment]
Can you try if this patch works?

```diff
diff --git a/manifoldDeps.cmake b/manifoldDeps.cmake
index 182145b..47e6cb5 100644
--- a/manifoldDeps.cmake
+++ b/manifoldDeps.cmake
@@ -1,8 +1,8 @@
 include(FetchContent)
 include(GNUInstallDirs)
-find_package(PkgConfig)
+find_package(PkgConfig QUIET)
 if(MANIFOLD_PAR STREQUAL "TBB")
-    find_package(TBB)
+    find_package(TBB QUIET)
 endif()
 if (PKG_CONFIG_FOUND)
     pkg_check_modules(Clipper2 Clipper2)
@@ -37,15 +37,15 @@ else()
     endif()
 endif()
 
-FetchContent_Declare(glm
-    GIT_REPOSITORY https://github.com/g-truc/glm.git
-    GIT_TAG b06b775c1c80af51a1183c0e167f9de3b2351a79
-    GIT_PROGRESS TRUE
-    FIND_PACKAGE_ARGS NAMES glm
-)
-FetchContent_MakeAvailable(glm)
-
+find_package(glm QUIET)
 if(NOT glm_FOUND)
+    set(GLM_BUILD_INSTALL "ON" CACHE STRING "install glm")
+    FetchContent_Declare(glm
+        GIT_REPOSITORY https://github.com/g-truc/glm.git
+        GIT_TAG b06b775c1c80af51a1183c0e167f9de3b2351a79
+        GIT_PROGRESS TRUE
+    )
+    FetchContent_MakeAvailable(glm)
     message(STATUS "glm not found, downloading from source")
     if(NOT EMSCRIPTEN)
         install(TARGETS glm EXPORT glmTargets)
diff --git a/src/utilities/CMakeLists.txt b/src/utilities/CMakeLists.txt
index bdd2253..46c463a 100644
--- a/src/utilities/CMakeLists.txt
+++ b/src/utilities/CMakeLists.txt
@@ -60,7 +60,7 @@ endif()
 target_include_directories(${PROJECT_NAME} INTERFACE
     $<INSTALL_INTERFACE:include/${CMAKE_PROJECT_NAME}>
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
-target_link_libraries(${PROJECT_NAME} INTERFACE glm::glm)
+target_link_libraries(${PROJECT_NAME} INTERFACE $<BUILD_INTERFACE:glm::glm>)
 
 if(NOT DEFINED thrust_SOURCE_DIR)
     set(thrust_SOURCE_DIR ${_THRUST_INCLUDE_DIR})
```

**Comment by howardt** [comment]
Yes!
It works fine with that patch.
